components/ca-vote.js

```diff
@@ -0,0 +1,156 @@
+/* global api */
+define(['./helpers/create-element.js', './helpers/clear-element.js', 'document-register-element'], (createElement, clearElement) => {
+    const componentName = 'ca-vote';
+
+    /**
+     * Class Vote for a ca-vote web component.
+     * This component is used to allow users to vote for prospective features in an application.
+     * @extends HTMLElement
+     */
+    class Vote extends HTMLElement {
+
+        /**
+         * createdCallback - called when the component created, (not yet attached to the DOM).
+         * @returns {undefined} initalises the component.
+         */
+        createdCallback() {
+            this._storageKey = `${componentName}_${this.id}`;
+            this.render();
+        }
+
+        /**
+         * observedAttributes - get a list of attributes that should effect re-render.
+         * @return {Array} list of attributes that effect a re-render.
+         */
+        static get observedAttributes() {
+            return ['id'];
+        }
+
+        /**
+         * @description Executes when any attribute is changed on the element
+         * @type {Event}
+         * @param {string} attrName - the name of the attribute to have changed
+         * @param {string} oldVal - the old value of the attribute
+         * @param {string} newVal - the new value of the attribute
+         * @returns {void} void
```

Comment:
I suppose technically it does return undefined. Although http://stackoverflow.com/questions/4759175/how-to-return-void-in-jsdoc.